src/hotspot/os/aix/os_aix.cpp

```diff
@@ -1159,6 +1159,33 @@ void *os::dll_load(const char *filename, char *ebuf, int ebuflen) {
   }
   return nullptr;
 }
+/*
+ Load library named <filename>
+ dll_build_name constructs the entire path as libfilename.so
+ Search order:
+ Load "libfilename.so" first,then load libfilename.a,on failure.
+ In,OpenJ9,the libary with .so extension is loaded first and then .a extension,on failure.
+*/
+void *os::dll_load(const char *filename, char *ebuf, int ebuflen) {
+  void* result = nullptr;
+  unsigned long buffer_length = strlen(filename);
+  int extension_length = 3;
+  char* file_path = NEW_C_HEAP_ARRAY(char, buffer_length + extension_length + 1, mtInternal);
+  strncpy(file_path,filename, buffer_length + 1);
+  char* const pointer_to_dot = strrchr(file_path, '.');
+  assert(pointer_to_dot != nullptr, "Attempting to load a shared object without extension? %s", filename);
```

Comment:
This should not only be an assertion. I think the check could be used instead of the strcmp below.

Reply:
I didn't follow that. You mean i need to keep a check if it is null and print it out ?

Reply:
An assertion is only used for debug builds. Such an error should be handled in product builds as well. I think an attempt to load an invalid library should simply fail. You may add logging if needed.
@tstuefe: Do you agree or have another proposal to handle such errors?

Reply:
In addition, the nullptr check is important to avoid undefined behavior when passing `pointer_to_dot` to any string function.

Reply:
Ok. So then may be we can return a nullptr and do a   log_info(os) with the correct error report ? @tstuefe @TheRealMDoerr

Reply:
I'd probably use `log_info(library)`.